src/permissions/NativeTokenRollingSpendLimit/NativeTokenRollingSpendLimitPermission.sol

```diff
@@ -88,10 +105,21 @@ contract NativeTokenRollingSpendLimitPermission is IPermissionContract {
                 if (call.target != address(this) || keccak256(call.data) != keccak256(assertSpendData)) {
                     revert MissingAssertSpend();
                 }
-            } else if (bytes4(call.data) == IPermissionCallable.permissionedCall.selector) {
+            } else if (selector == IPermissionCallable.permissionedCall.selector) {
                 // check call target is the allowed contract
                 // assume PermissionManager already prevents account as target
                 if (call.target != allowedContract) revert UserOperationUtils.TargetNotAllowed();
+            } else if (calls[i].target == MAGIC_SPEND_ADDRESS) {
```

Comment:
it should always probably be the first call, if we want to leverage that

Reply:
The first call is used for our execution phase manager checks, but maybe could force this to be second call